XLS-0075d-account-permission-delegation/README.md

```diff
@@ -0,0 +1,314 @@
+
+### 2.2. Account Deletion
+
+The `Delegate` object is not a [deletion blocker](https://xrpl.org/docs/concepts/accounts/deleting-accounts/#requirements).
+
+## 3. Transaction: `DelegateSet`
+
+This transaction allows an account to delegate certain permissions to another account. It is loosely modeled to be similar to the [`DepositPreauth` transaction type](https://xrpl.org/docs/references/protocol/transactions/types/depositpreauth).
+
+### 3.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`TransactionType`| ✔️|`string`|`UInt16`|The transaction type (`DelegateSet`).|
+|`Account`| ✔️|`string`|`AccountID`|The account that wants to authorize another account.|
+|`Authorize`|✔️|`string`|`AccountID`|The authorized account.|
+|`Permissions`| ✔️|`string`|`STArray`|The transaction permissions that the account has been granted.|
+
+### 3.1.1. `Permissions`
+
+This transaction works slightly differently from the `DepositPreauth` transaction type. Instead of using an `Unauthorize` field, an account is unauthorized by using an empty `Permissions` list.
+
+### 3.2. Failure Conditions
+
+* `Permissions` is too long (the limit is 10), or includes duplicates.
+* Any of the specified permissions are invalid.
+* `Authorize` is the same as `Account`.
+
+### 3.3. State Changes
+
+A `Delegate` object will be created, modified, or deleted based on the provided fields.
+
+## 4. Transactions: Common Fields
+
+### 4.1. Fields
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/transactions/common-fields/) are the fields that all transactions currently have.
+
+<!--There are too many and I didn't want to list them all, it cluttered up the spec - but maybe it can be a collapsed section?-->
+
+We propose these modifications:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Delegate`| |`string`|`AccountID`|The delegate account that is sending the transaction.|
+
+#### 4.1.1. `Delegate`
+
+The way the `Delegate` field works is somewhat akin to `RegularKey`. The `Account` field is the delegating account, the `Delegate` field is the delegate, and the public key and signature is based on the delegate's keys.
+
+The delegate will pay the fees on the transaction, to prevent a delegate from draining an account's XRP via fees.
+
+### 4.2. Failure Conditions
+
+* The `Account` hasn't authorized the `Delegate` to send transactions on behalf of it.
+* The `Account` hasn't authorized the `Delegate` to send this particular transaction type/granular permission on behalf of it.
+* `Delegate` is the same as `Account`.
+
+### 4.3. State Changes
+
+The transaction succeeds, and the fee is charged to the `Delegate` account. 
+
+## 5. Examples
```

Comment:
It will be helpful if you can explicitly document the changes to the `account_tx` RPC. Based on my experiments, the `delegate account` also indexes the transactions which are sent on behalf of the `delegating account`. Since there are two different accounts involved in all the transactions, this behavior is not obvious at the outset.

You can look at the tip of [this branch](https://github.com/XRPLF/xrpl-py/compare/main...ckeshava:xrpl-py:xls-74-sequence?expand=1) (or) [this commit](https://github.com/XRPLF/xrpl-py/commit/19639e78c4d8d43fe6f51efbb14d37bc349ffe58) for the specifics.

Reply:
On that note, it will be helpful to have additional AccountTx-related tests. I did not come across any such tests [in this PR](https://github.com/XRPLF/rippled/pull/5354). This is [the relevant file](https://github.com/XRPLF/rippled/blob/develop/src/test/rpc/AccountTx_test.cpp)

Reply:
Changes to `account_tx` do not need to be explicitly made, therefore they don't need to be explicitly called out in the spec.